# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [6]:
# TODO
import scipy.stats as stats 
import numpy as np 

diet1 = np.array([2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5])
diet2 = np.array([3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2])

t_stat, p_value = stats.ttest_ind(diet1, diet2)

print(f"Estadístico t: {t_stat}")
print(f"P-valor: {p_value}")


if p_value < 0.05:
    print("Conclusión: Se rechaza la Hipótesis Nula (Ho).")
    print("Justificación: El p-valor es menor al 5%, lo que indica que la "
          "diferencia observada no es casualidad. Las dietas son distintas.")
else:
    print("Conclusión: No hay evidencia suficiente para rechazar Ho.")
    print("Justificación: La diferencia podría deberse al azar.")


Estadístico t: -3.5383407969933938
P-valor: 0.004083270191713912
Conclusión: Se rechaza la Hipótesis Nula (Ho).
Justificación: El p-valor es menor al 5%, lo que indica que la diferencia observada no es casualidad. Las dietas son distintas.


### Conclusión 

Dado que el objetivo es determinar si una dieta es más efectiva que la otra, se establece como hipótesis nula ($H_0$) que no existe diferencia entre la pérdida de peso promedio de la Dieta 1 y la Dieta 2 (m1= m2). Al realizar la prueba t de Student, se obtuvo un p-valor de 0.004. Al ser este valor menor al nivel de significancia de 0.05, se rechaza la hipótesis nula. Esto confirma que existe una diferencia estadísticamente significativa entre las medias de ambas dietas, siendo la Dieta 2 más efectiva en esta muestra.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [13]:
# TODO

import numpy as np
import scipy.stats as stats 

fert_1 = np.array([20, 21, 20, 19, 20])
fert_2 = np.array([22, 21, 23, 22, 21])
fert_3 = np.array([24, 23, 22, 23, 24])

f_stat, p_value = stats.f_oneway(fert_1, fert_2, fert_3)

print(f"Estadístico t: {f_stat}")
print(f"P-valor: {p_value}")

alpha = 0.05 

if p_value < 0.05:
    print("Conclusión: Se rechaza la Hipótesis Nula (Ho).")
    print("Justificación: El p-valor es menor al 5%, lo que indica que la "
          "diferencia observada no es casualidad. Los fertilizantes son distintos.")
else:
    print("Conclusión: No hay evidencia suficiente para rechazar Ho.")
    print("Justificación: La diferencia podría deberse al azar.")

import numpy as np
from statsmodels.stats.multicomp import pairwise_tukeyhsd

data = np.concatenate([fert_1, fert_2, fert_3])
labels = ["Fert1"] * 5 + ["Fert2"] * 5 + ["Fert3"] * 5

# Tukey test
result = pairwise_tukeyhsd(data, labels, alpha = 0.05)
print(result)




Estadístico t: 20.31578947368421
P-valor: 0.000140478247931904
Conclusión: Se rechaza la Hipótesis Nula (Ho).
Justificación: El p-valor es menor al 5%, lo que indica que la diferencia observada no es casualidad. Los fertilizantes son distintos.
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
 Fert1  Fert2      1.8 0.0099 0.4572 3.1428   True
 Fert1  Fert3      3.2 0.0001 1.8572 4.5428   True
 Fert2  Fert3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


### Conclusión 

Podemos saberlo mediante una Prueba Post-Hoc de Tukey. Los datos muestran que el Fertilizante 3 es el mejor de todos, ya que supera al Fertilizante 2 por 1.4 kg y al Fertilizante 1 por 3.2 kg en promedio. Todas estas diferencias son estadísticamente significativas (p < 0.05), por lo que no son producto de la casualidad